clinica/pipelines/dwi_preprocessing_using_t1/dwi_preprocessing_using_t1_utils.py

```diff
@@ -160,36 +160,34 @@ def rotate_bvecs(in_bvec, in_matrix):
     return out_file
 
 
-def ants_combine_transform(fix_image, moving_image, ants_warp_affine):
+def ants_apply_transform_warp_image(fix_image, moving_image, ants_warp_affine):
     import os
+    import subprocess
 
-    out_warp_field = os.path.abspath("out_warp_field.nii.gz")
     out_warped = os.path.abspath("out_warped.nii.gz")
 
     cmd = (
-        f"antsApplyTransforms -o [out_warp_field.nii.gz,1] -i {moving_image} -r {fix_image} "
-        f"-t {ants_warp_affine[0]} {ants_warp_affine[1]} {ants_warp_affine[2]}"
-    )
-    os.system(cmd)
-
-    cmd1 = (
-        f"antsApplyTransforms -o out_warped.nii.gz -i {moving_image} -r {fix_image} "
-        f"-t {ants_warp_affine[0]} {ants_warp_affine[1]} {ants_warp_affine[2]}"
+        f"antsApplyTransforms -o {out_warped} -i {moving_image} -r {fix_image} "
+        f"-t {ants_warp_affine[0]} -t {ants_warp_affine[1]} -t {ants_warp_affine[2]}"
     )
-    os.system(cmd1)
+    subprocess.run(cmd, shell=True)
 
-    return out_warp_field, out_warped
+    return out_warped
 
 
-def create_jacobian_determinant_image(imageDimension, deformationField, outputImage):
+def ants_apply_transform_warp_field(fix_image, moving_image, ants_warp_affine):
     import os
+    import subprocess
 
-    outputImage = os.path.abspath(outputImage)
+    out_warp_field = os.path.abspath("out_warp_field.nii.gz")
 
-    cmd = f"CreateJacobianDeterminantImage {str(imageDimension)} {deformationField} {outputImage}"
-    os.system(cmd)
+    cmd = (
+        f"antsApplyTransforms -o [{out_warp_field},1] -i {moving_image} -r {fix_image} "
```

Comment:
I understand the idea of creating two Nipype nodes in case the interface is used again. But for the moment it just seems we are redundantly calling the same command twice. I think a comment in the code to justify this would be useful.

Reply:
> calling the same command twice

It's not the *same* command. Granted both share the same set of inputs, but output differs (warped image vs deformation field).

> the idea of creating two Nipype nodes in case the interface is used again

Using two nodes puts us closer to how we would have used the actual nipype interface for `antsApplyTransforms` if it were working as intended. More importantly, this setup also gives nipype more freedom to schedule workflow execution, since both transforms can be run in parallel (same inputs, different outputs).

Reply:
I've added another comment in the workflow part. I've also factorized this function. Do you think it is clearer ?

Reply:
Ok, I think I have got an idea that may satisfy everyone:

Reply:
let's define a function with the following signature:

```python
def antsApplyTransform(
    fixed_image: str,
    moving_image: str,
    transforms: List[str],
    warped_image: str,
    output_warped_image: bool = True
) -> None:
    import subprocess

    # Whether we want the warped image or transformation field as output.
    output = f"{warped_image}" if output_warped_image else f"[{warped_image}, 1]"
    
    # Common template for the command.
    cmd = (
        f"antsApplyTransforms -o {output} -i {moving_image} -r {fixed_image} "
        f"-t {transforms[0]} -t {transforms[1]} -t {transforms[2]}"
    )

    # Perform the actual call.
    subprocess.run(cmd, shell=True)
```

Wrap it in a Function node and set `warped_image` and `output_warped_image` accordingly when constructing the workflow. That would put us very close to the nipype interface, naming-wise and functionality-wise, whilst making a future replacement by `nipype.interfaces.ants.Registration` seamless.

Reply:
All right, so what I was missing is that the "output" is actually an argument that defines the behavior of the function. The proposal looks good.

Reply:
Code updated with missing absolute path conversion.